server_test.go

```diff
@@ -208,6 +208,51 @@ func TestOpenStatRace(t *testing.T) {
 	checkServerAllocator(t, server)
 }
 
+func TestOpenWithPermissions(t *testing.T) {
+	client, server := clientServerPair(t)
+	defer client.Close()
+	defer server.Close()
+
+	tmppath := path.Join(os.TempDir(), "open_permissions")
+	pflags := flags(os.O_RDWR | os.O_CREATE | os.O_TRUNC)
+	ch := make(chan result, 2)
+	id1 := client.nextID()
+	id2 := client.nextID()
+
+	// New files should have their permissions set.
+	client.dispatchRequest(ch, &sshFxpOpenPacket{
+		ID:     id1,
+		Path:   tmppath,
+		Pflags: pflags,
+		Flags:  sshFileXferAttrPermissions,
+		Attrs:  []byte{0x0, 0x0, 0x1, 0xe5}, // 0o745 -- a slightly strange permission to test.
+	})
+	<-ch
+	stat, err := os.Stat(tmppath)
+	assert.NoError(t, err)
+	if !assert.Equal(t, os.FileMode(0o745), stat.Mode()&os.ModePerm) {
+		t.Logf("stat.Mode() = %v", stat.Mode())
+	}
+
+	// Existing files should not have their permissions changed.
+	client.dispatchRequest(ch, &sshFxpOpenPacket{
+		ID:     id2,
+		Path:   tmppath,
+		Pflags: pflags,
+		Flags:  sshFileXferAttrPermissions,
+		Attrs:  []byte{0x0, 0x0, 0x1, 0xed}, // 0o755
+	})
+	<-ch
+	stat, err = os.Stat(tmppath)
+	assert.NoError(t, err, "mode should not have changed")
```

Comment:
This test failure message doesn’t make sense. We don’t know if the mode has changed or not yet. We just know that the `Stat` failed, and this could point to a deeper issue than just “mode should not have changed“.

Fix: just don’t put a log message here. Simply there being an error message popping up here is clue enough that an unexpected error occurred, and deeper review is necessary of what is happening.